ring_buffer/ring_buffer.py

```diff
@@ -4,8 +4,63 @@ def __init__(self, capacity):
     self.current = 0
     self.storage = [None]*capacity
 
+  
+
   def append(self, item):
+# self.current is the newest element. Old is characterized by an element to
```

Comment:
I see you got it right, but it can be cleaned up a bit. You have a few extra things that you don't really need and you are repeating yourself in a few place. A few things to remember, first we always have access to the last index with `self.current`. Second, it is okay to override the current item, because if we are at capacity, then we will start at the beginning.

```   
self.storage[self.current] = item
self.current += 1

if self.current == self.capacity:
  self.current = 0
```

if you have that, it satisfies what we need it to do. If you need to me explain some more let me know. Otherwise take a look at your code again and see where you are repeating your self where you don't need to